Flaky test: test_that_triangular_is_symmetric_around_mode

**Describe the bug**

```
FAILED tests/unit_tests/config/test_transfer_functions.py::test_that_triangular_is_symmetric_around_mode - assert 1.52587890625e-05 < 1e-05
 +  where 1.52587890625e-05 = abs(((137438953472.0 - 137438953471.7071) - (137438953472.2929 - 137438953472.0)))
Falsifying example: test_that_triangular_is_symmetric_around_mode(
    args=(137438953472.0, 137438953471.0, 137438953473.0),
)
```

Observed in https://github.com/equinor/ert/actions/runs/7624520284/job/20766888361?pr=6998

Another occurence:
```
FAILED tests/unit_tests/config/test_transfer_functions.py::test_that_triangular_is_symmetric_around_mode - assert 3.0517578125e-05 < 1e-05
 +  where 3.0517578125e-05 = abs(((-254884719877.7367 - -254884719878.0296) - (-254884719877.44382 - -254884719877.7367)))
Falsifying example: test_that_triangular_is_symmetric_around_mode(
    args=(-254884719877.7367, -254884719878.7367, -254884719876.73666),
)

You can reproduce this example by temporarily adding @reproduce_failure('6.96.4', b'AXicY2BgbIpdoyHWNNOWAcpsloQzmzJtAYz0CLg=') as a decorator on your test case
```

My lucky day:
```
FAILED tests/unit_tests/config/test_transfer_functions.py::test_that_triangular_is_symmetric_around_mode - assert 0.0001220703125 < 1e-05
 +  where 0.0001220703125 = abs(((999999999999.0 - 999999999998.4143) - (999999999999.5858 - 999999999999.0)))
Falsifying example: test_that_triangular_is_symmetric_around_mode(
    args=(999999999999.0, 999999999997.0001, 1000000000000.9998),
)

You can reproduce this example by temporarily adding @reproduce_failure('6.96.4', b'AAAAAAAAAOjUpQ//AAAAgn0alKH/sAAAAACCfRqUogAP/w==') as a decorator on your test case
```

--- a/tests/unit_tests/config/test_transfer_functions.py
+++ b/tests/unit_tests/config/test_transfer_functions.py
@@ -191,7 +191,9 @@ def test_that_triangular_is_symmetric_around_mode(args):
     y2 = TransferFunction.trans_triangular(x2, [_min, _mode, _max])
 
     # Check if y1 and y2 are symmetric around the mode
-    assert abs((_mode - y1) - (y2 - _mode)) < 1e-5
+    assert abs((_mode - y1) - (y2 - _mode)) < 1e-15 * max(
+        *map(abs, [x1, x2, _min, _mode, _max])
+    )
 
 
 @given(valid_triangular_params())

